nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ndarray/BaseNDArray.java

```
@@ -3810,6 +3826,18 @@ public int rank() {
      */
     @Override
     public int length() {
+        if(length >= Integer.MAX_VALUE)
+            throw new IllegalArgumentException("Length can not be >= Integer.MAX_VALUE");
```

Maybe return a different error message here, to avoid confusing users. Length *can* be >= Integer.MAX_VALUE internally, we just can't return it using the length() method.